azure/services/async/async.go

```diff
@@ -183,12 +197,27 @@ func (s *Service) DeleteResource(ctx context.Context, spec azure.ResourceSpecGet
 	return nil
 }
 
-// retryAfter returns the max between the `RETRY-AFTER` header and the default requeue time.
+// getRequeueAfterFromFuture returns the max between the `RETRY-AFTER` header and the default requeue time.
 // This ensures we respect the retry-after header if it is set and avoid retrying too often during an API throttling event.
-func retryAfter(sdkFuture azureautorest.FutureAPI) time.Duration {
+func getRequeueAfterFromFuture(sdkFuture azureautorest.FutureAPI) time.Duration {
 	retryAfter, _ := sdkFuture.GetPollingDelay()
 	if retryAfter < reconciler.DefaultReconcilerRequeue {
 		retryAfter = reconciler.DefaultReconcilerRequeue
 	}
 	return retryAfter
 }
+
+// getRetryAfterFromDetailedError returns the time.Duration from the http.Response in the autorest.DetailedError.
+// If there is no Response object, or if there is no meaningful Retry-After header data, we return a zero value time.Duration.
+func getRetryAfterFromDetailedError(detailedError autorest.DetailedError) (ret time.Duration) {
+	if detailedError.Response != nil {
+		if retryAfter := detailedError.Response.Header.Get("Retry-After"); retryAfter != "" {
+			if rai, _ := strconv.Atoi(retryAfter); rai > 0 {
+				ret = time.Duration(rai) * time.Second
```

Comment:
why are we doing * time.Second here ?

Reply:
See my latest changes, which account for both flavors of `Retry-After` data. Ref:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After#examples